crates/holochain/src/core/ribosome/host_fn/trace.rs

```diff
@@ -4,54 +4,74 @@ use crate::core::ribosome::RibosomeT;
 use holochain_types::prelude::*;
 use std::sync::Arc;
 use tracing::*;
+use once_cell::unsync::Lazy;
 
-#[instrument(skip(_ribosome, _call_context, input))]
-pub fn debug(
+#[instrument(skip(input))]
+pub fn wasm_trace(
+    input: TraceMsg,
+) {
+    match input.level {
+        holochain_types::prelude::Level::TRACE => tracing::trace!("{}", input.msg),
+        holochain_types::prelude::Level::DEBUG => tracing::debug!("{}", input.msg),
+        holochain_types::prelude::Level::INFO => tracing::info!("{}", input.msg),
+        holochain_types::prelude::Level::WARN => tracing::warn!("{}", input.msg),
+        holochain_types::prelude::Level::ERROR => tracing::error!("{}", input.msg),
+    }
+}
+
+pub fn trace(
     _ribosome: Arc<impl RibosomeT>,
     _call_context: Arc<CallContext>,
-    input: DebugMsg,
+    input: TraceMsg,
 ) -> RibosomeResult<()> {
-    debug!(
-        "{}:{}:{} {}",
-        input.module_path(),
-        input.file(),
-        input.line(),
-        input.msg()
-    );
+    // Avoid dialing out to the environment on every trace.
+    let wasm_log = Lazy::new(||{
+        std::env::var("WASM_LOG").unwrap_or_else(|_| "[wasm_trace]=debug".to_string())
+    });
+    let collector = tracing_subscriber::fmt()
+    .with_env_filter(tracing_subscriber::EnvFilter::new((*wasm_log).clone()))
+    .with_target(false)
+    .finish();
+    tracing::subscriber::with_default(collector, || {
+        wasm_trace(input)
+    });
```

Comment:
Is this logic just needed to have separate WASM_LOG vs RUST_LOG?

Reply:
@neonphog that's part of yes, to allow for differentiating the binary tracing vs. happ tracing, but also:

- we need `with_target` set to false to prevent the host debug information bleeding into the guest output
- the observability crate is not instrumented at this point, so if we do nothing here then the default experience of a happ developer is no output, they have to learn how instrument observability and use it before they can do any tracing work

we can tweak this in the future if the ergonomics don't feel right, e.g. like we discussed last night on the call we could hook it into observability by retooling or extending this here, then we'd have visibility at the network diagnostics level of what is happening inside wasm, which could be cool or could be a waste of time :shrug: 

cc: @zippy @freesig